@here/harp-utils/lib/ObjectUtils.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * Deep clone of object.
+ *
+ * Like `JSON.parse(JSON.stringify(obj))`, but supports basic javascript types (string, number,
+ * object), `Date` and `RegExp`s and cycles.
+ *
+ * @param obj
+ */
+export function cloneDeep<T>(obj: T): T {
```

Comment:
Are there any known limitations, especially those that would make it fail horribly? If no, great, if yes, please add comments and maybe test(s).
This may be important if the theme is no longer fully loaded from JSON, but may have been modified by users.